File: Stack.js
class Stack {
  constructor() {
    this.stack = new Array();
  }

  push(ele) {
    this.stack.push(ele);
  }

  pop() {
    return this.stack.pop();
  }

  peek() {
    let lastIdx = this.stack.length - 1;
    return this.stack[lastIdx];
  }
}

module.exports = Stack;

File: stack.rb
# stack abstract data type implemented with an array
# LIFO

class Stack
  def initialize
    @stack = Array.new
  end

  def push(ele)
    @stack.push(ele)
  end

  def pop
    @stack.pop
  end

  def peek
    @stack.last
  end

  def size
    @stack.size
  end
end

